.github/workflows/main.yml

```diff
@@ -3,53 +3,81 @@ name: main
 on:
   push:
   pull_request:
-    branches: [ main ]
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
   php-cs-fixer:
     runs-on: ubuntu-latest
     name: PHP CS FIXER
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: PHP-CS-Fixer
         uses: docker://oskarstark/php-cs-fixer-ga
         with:
           args: --diff --dry-run
-  phpunit:
+
+  integration:
     runs-on: ${{ matrix.os }}
+    needs:
+      - php-cs-fixer
     strategy:
+      fail-fast: false
       matrix:
         php: [8.1, 8.2, 8.3]
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        dependencies: [ 'lowest', 'highest' ]
-    continue-on-error: ${{ matrix.os == 'windows-latest' && matrix.php != '8.1' }}
-    name: PHP ${{ matrix.php }} - OS ${{ matrix.os }} - Dependencies ${{ matrix.dependencies }}
+        os: [ubuntu-latest, macos-latest]
+        include:
+          - os: windows-latest
+            php: 8.1
```

Comment:
I Love everything apart from turning off windows over rar files

Reply:
So I think let remove rar support for now. We can add it again when it's well supported